scripts/zones/The_Shrine_of_RuAvitau/mobs/Kirin_s_Avatar.lua

```diff
@@ -0,0 +1,80 @@
+-----------------------------------
+-- Area: The Shrine of Ru'Avitau
+-- NPC:  Kirin's Avatar: 17506675
+-----------------------------------
+package.loaded[ "scripts/zones/The_Shrine_of_RuAvitau/TextIDs" ] = nil;
+-----------------------------------
+
+require( "scripts/zones/The_Shrine_of_RuAvitau/TextIDs" );
+require( "scripts/globals/status" );
+
+-----------------------------------
+-- onMobInitialize Action
+-----------------------------------
+function onMobInitialize( mob )
+    
+end
+
+-----------------------------------
+-- onMobSpawn Action
+-----------------------------------
+function onMobSpawn( mob )
+    mob:setModelId(math.random(791, 798));
+    mob:hideName(false);
+    mob:untargetable(true);
+    mob:setUnkillable(true);
+end
+
+-----------------------------------
+-- onMobFight
+-- Actions:
+--  1 = Attacking
+-- 34 = Preparing Astral Flow
+-- 26 = Astral Flow used
+-----------------------------------
+function onMobFight( mob, target )
+    local id = mob:getID();
```

Comment:
I believe I tried calling mob:useMobAbility in onMobSpawn (didn't work), but I will try once more to be absolutely sure.